EKF/RingBuffer.h

```diff
@@ -158,6 +158,29 @@ class RingBuffer
 		return false;
 	}
 
+	bool read_first_older_than(const uint64_t& timestamp, data_type *sample)
```

Comment:
How have you verified that this new function is retrieving the correct data? The system level effect of retrieving data at the wrong time will be subtle and could easily be missed.